lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -207,6 +210,12 @@ export abstract class ClientApplication {
             this.logger.verbose("NavigateToLoginRequestUrl set to false, handling hash");
             return this.handleHash(responseHash, state);
         } else if (!BrowserUtils.isInIframe()) {
+            if (typeof this.clientSideNavigateCallback === "function") {
+                const urlParts = new UrlString(loginRequestUrl).getUrlComponents();
+                await this.clientSideNavigateCallback(urlParts.AbsolutePath, urlParts.QueryString, urlParts.Hash);
+                this.logger.verbose("clientSideNavigate completed navigation, handling hash");
+                return this.handleHash(responseHash, state);
```

Comment:
This feels too locked to this specific feature. Could it be baked into `BrowserUtils.navigateWindows`?